docs/installing-on-gke.md

```diff
@@ -0,0 +1,321 @@
+
+## Step 5: Setup dfTimewolf and CLI configs
+
+OSDFIR Infrastructure utilizes dfTimewolf for orchestrating forensic collection
+and processing. dfTimewolf allows you to define "recipes" that specify how data
+should be collected, processed by tools like Turbinia, and exported to platforms
+like Timesketch.
+
+To install dfTimewolf, you'll need to have Python 3.11 or greater, `git`, and
+`pip` installed on your machine. dfTimewolf uses Poetry for simplified dependency
+management. Follow these steps to install dfTimewolf:
+
+```bash
+git clone https://github.com/log2timeline/dftimewolf.git && cd dftimewolf
+pip install poetry
+poetry install && poetry shell
+```
+
+Retrieve the Timesketch password from your deployment. For example, to grab
+it from a release named `my-release`, run:
+
+```bash
+kubectl get secret --namespace default my-release-timesketch-secret -o jsonpath="{.data.timesketch-user}" | base64 -d
+```
+
+dfTimewolf uses a configuration file called `.dftimewolfrc` to store settings
+such as your Timesketch credentials and endpoint. This allows you to avoid
+entering these details every time you run a recipe.
+
+Now, create a `.dftimewolfrc` file in your HOME directory, replacing
+`$TIMESKETCH_PASSWORD` with the Timesketch password retrieved in the previous step:
+
+```bash
+cat >> ~/.dftimewolfrc << EOF
+{
+"timesketch_username": "timesketch",
+"timesketch_password": "$TIMESKETCH_PASSWORD",
+"timesketch_endpoint": "http://127.0.0.1:5000",
+"turbinia_api": "http://127.0.0.1:8000"
+}
+EOF
+```
+
+Now you have dfTimewolf installed and configured to interact with your OSDFIR
+Infrastructure deployment.
+
+## Step 6: Process a Google Cloud Disk
+
+With OSDFIR Infrastructure deployed and dfTimewolf installed and configured,
+you're ready to process a GCP disk.
+
+This example uses the dfTimewolf `gcp_turbinia_ts` recipe, which processes an
+existing GCP persistent disk with Turbinia and sends the resulting Plaso timeline
+to Timesketch.
+
+First, create a disk to process using a name such as `test-disk`:
+
+```bash
+gcloud compute disks create test-disk --zone $ZONE
+```
+
+> *Important*: The recipe requires that the disk being processed
+is in the same zone Turbinia is deployed to.
+
+You'll need to use `kubectl port-forward` to forward the Turbinia and Timesketch
+services locally to your machine. This allows you to access the Turbinia UI and
+the Timesketch API from your local machine.
+
+For example, to port-forward from a release named `my-release`, run the following
+commands in two seperate terminals:
+
+```bash
+kubectl --namespace default port-forward service/my-release-turbinia 8000:8000 
+kubectl --namespace default port-forward service/my-release-timesketch 5000:5000  
+```
+
+This will allow dfTimewolf to access the Turbinia and Timesketch services locally
+from your machine.
+
+Then on a third terminal, run the dfTimewolf recipe:
+
+```bash
+dftimewolf gcp_turbinia_ts $PROJECT_ID --disk_names test-disk
```

Comment:
You need to add $ZONE at the end iof the command.
`dftimewolf gcp_turbinia_ts $PROJECT_ID --disk_names test-disk $ZONE`

Reply:
Command (and all Turbinia tasks failed) with below error:
````
("Failed to retrieve http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/turbinia@ramses-test5.iam.gserviceaccount.com/token?scopes=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloud-platform%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcompute%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcompute.readonly%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdevstorage.full_control%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdevstorage.read_only%2Chttps%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdevstorage.read_write from the Google Compute Engine metadata service. Status: 404 Response:\nb'Unable to generate access token; IAM returned 404 Not Found: Not found; Gaia id not found for email turbinia@ramses-test5.iam.gserviceaccount.com\\n'", <google_auth_httplib2._Response object at 0x7e88f6045de0>). Something is wrong with your Application Default Credentials. Try running: $ gcloud auth application-default login]
```

Reply:
This looks like some service account/permission issue.....

I tried everything in a fresh/clean/reset gcp cloudshell (so a new vm)

Reply:
Ack thanks fixed by adding the ZONE.

Regarding the Turbinia service account not working... I realize that was an issue on my end not pushing the code. Basically I've updated the docs to follow this method https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#configure-authz-principals

Instead of https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#kubernetes-sa-to-iam

So what happened was it was still annotating the code with the GCP annotation when it doesn't need it anymore and it caused the permission issues you saw. Doing it in the new method prevents us from having to annotate the K8s service account and re-updating the annotation on the GKE cluster itself which IMO has more issues to go wrong. 

I will push some changes and link you what you need for it to work.